recipe/meta.yaml

```diff
@@ -1,17 +1,17 @@
 {% set name = "networkx" %}
-{% set version = "2.8.4" %}
+{% set version = "3.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 5e53f027c0d567cf1f884dbb283224df525644e43afd1145d64c9d88a3584762
+  sha256: de346335408f84de0eada6ff9fafafff9bcda11f0a0dfaa931133debb146ab61
 build:
   number: 1
   skip: True  # [py<38]
-  script: {{ PYTHON }} -m pip install . -vvv --no-deps
+  script: {{ PYTHON }} -m pip install . -vvv --no-deps --no-build-isolation
 
 requirements:
   host:
```

Comment:
You could add a run_constrained section with the content of:
https://github.com/networkx/networkx/blob/networkx-3.1/requirements/default.txt and https://github.com/networkx/networkx/blob/networkx-3.1/requirements/extra.txt

Reply:
And remove the big comment form the run section.

Reply:
Ok I've made those changes

Reply:
Why do we need to move all requirements to the `run_constrained` section? Perhaps my understanding of `run_constrained` is limited.

Reply:
If the user will install a package mentioned in `run_constrained` to the  environment then the package will have a the range of the pinnings from `run_constrained`

Reply:
this way they are optional dependencies, and the constraint will still ensure that they are in the valid range if the user install the optional dependency manually

Reply:
I understand now what run_constrained does, but I'm still not convinced by the fact that packages that are listed as default requirements are considered optional dependencies. You can convince me offline if you want.

Reply:
the thing is that every dependency here is optional, even the ones in default.txt. you can see that install_requires is empty in setup.py: https://github.com/networkx/networkx/blob/networkx-3.1/setup.py#L167

To install the deps from default.txt with pip you'd use `pip install networkx[default]`

Reply:
@boldorider4 If you will add those dependencies to run then downstream packages will have a dependency hell. It has already happened with **v2.6.2** https://github.com/AnacondaRecipes/networkx-feedstock/commit/e0063b65a3d992664d96c3c0d275ef8052b2dee6 and the maintainers decided to remove all of them from runtime

Reply:
got it. Thanks everyone for clarifying this.